deepmd/descriptor/se_a.py

```diff
@@ -275,6 +289,30 @@ def get_ntypes(self) -> int:
         """
         return self.ntypes
 
+    def get_ntypes_spin(self) -> int:
+        """
+        Returns the number of atom types which contain spin
+        """
+        return self.ntypes_spin
+
+    def get_use_spin(self) -> List[bool]:
+        """
+        Returns the list of whether to use spin for each atom type
+        """
+        return self.use_spin
+
+    def get_spin_norm(self) -> List[float]:
+        """
+        Returns the list of magnitude of atomic spin for each atom type
+        """
+        return self.spin_norm
+
+    def get_virtual_len(self) -> List[float]:
+        """
+        Returns the list of distance between real atom and virtual atom for each atom type
+        """
+        return self.virtual_len
```

Comment:
Instead of creating the methods here, I suggest creating a new class `Spin`.

```py
class Spin:
    def __init__(self, ...):
        ...

    def get_ntypes_spin(self) -> int:
        ...
```

Then pass a class to the descriptor:
```py
spin = Spin(**spin_param)
descrpt_param['spin'] = spin
```

you can call
```py
self.spin.get_ntypes_spin()
```

Then we don't need to repeat it for every descriptor, and we are much easier to implement it for other descriptors.

Reply:
I see `spin` is in the `model` section. I am not sure if it is for descriptors only or like the type embedding.

Reply:
`spin` is used in both `DescrptSeASpin` and `model`. So, We have created class `spin` following @njzjz  suggestions.